pkg/envoy/secrets/secrets.go

```diff
@@ -63,3 +67,25 @@ func (sdsc *SDSCert) GetMeshService() (*service.MeshService, error) {
 		Name:      slices[1],
 	}, nil
 }
+
+// GetK8sServiceAccount unmarshals a K8sServiceAccount type from a SDSCert name
+func (sdsc *SDSCert) GetK8sServiceAccount() (*identity.K8sServiceAccount, error) {
+	slices := strings.Split(sdsc.Name, namespaceNameSeparator)
+	if len(slices) != 2 {
+		log.Error().Msgf("Error converting Service Account %s from string to K8sServiceAccount", sdsc.Name)
+		return nil, identity.ErrInvalidNamespacedServiceStringFormat
```

Comment:
Maybe move `ErrInvalidNamespacedServiceStringFormat` out of the `identity` package.

Reply:
Moved `ErrInvalidNamespacedServiceStringFormat` and `ErrInvalidMeshServiceFormat` to secrets/errors.go